kotlin/PixelPass/src/main/java/io/mosip/pixelpass/cose/KeyUtil.kt

```diff
@@ -0,0 +1,32 @@
+package io.mosip.pixelpass.cose
+
+import COSE.OneKey
+import net.i2p.crypto.eddsa.EdDSASecurityProvider
+import org.apache.commons.codec.binary.Base64
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import java.security.KeyFactory
+import java.security.Provider
+import java.security.Security
+import java.security.spec.KeySpec
+import java.security.spec.X509EncodedKeySpec
+
+object KeyUtil {
+    private val EDDSA_ALGO: String = "EdDSA"
+
+    @Throws(Exception::class)
+    fun oneKeyFromPublicKey(publicKeyStr: String?): OneKey {
+
+        //Why eddsa sec provider instead of bouncy castle?
```

Comment:
WHY?

Reply:
Seems like the implementation was done in hurry and specific to Philsys. This is what we discussed with Vishwa today.